app/code/community/Bolt/Boltpay/controllers/OrderController.php

```diff
@@ -61,13 +61,17 @@ public function saveAction()
             /////////////////////////////////////////////////////////
             /** @var Bolt_Boltpay_Helper_Transaction $transactionHelper */
             $transactionHelper = Mage::helper('boltpay/transaction');
-            /** @var  Bolt_Boltpay_Model_Order $orderModel */
+            /** @var Bolt_Boltpay_Model_Order $orderModel */
             $orderModel = Mage::getModel('boltpay/order');
             $order = $orderModel->getOrderByQuoteId($transactionHelper->getImmutableQuoteIdFromTransaction($transaction));
 
             if ($order->isObjectNew()) {
                 $sessionQuote = $checkoutSession->getQuote();
-                $orderModel->createOrder($reference, $sessionQuote->getId(), true, $transaction);
+                $sessionQuoteId = $sessionQuote->getId();
+                if (!$sessionQuoteId) {
+                    $sessionQuoteId = $immutableQuoteId = $transactionHelper->getImmutableQuoteIdFromTransaction($transaction);
```

Comment:
this is not a session quote. Also what happens if sessoinQuote exists but checkout is via product page checkout? Should we always use $immutableQuoteId here?